zcash_client_sqlite/src/wallet.rs

```diff
@@ -980,6 +1001,48 @@ pub(crate) fn block_height_extrema(
     })
 }
 
+pub(crate) fn get_account<P: Parameters>(
+    conn: &rusqlite::Connection,
+    params: &P,
+    account_id: AccountId,
+) -> Result<Option<Account>, SqliteClientError> {
+    let mut sql = conn.prepare_cached(
+        r#"
+        SELECT account_type, uvk, hd_seed_fingerprint, hd_account_index
+        FROM accounts
+        WHERE id = :account_id
+    "#,
+    )?;
+
+    let mut result = sql.query(params![account_id.0])?;
+    let row = result.next()?;
+    match row {
+        Some(row) => {
+            let account_type: u32 = row.get(0)?;
+            let uvk: String = row.get(1)?;
+            let ufvk = UnifiedFullViewingKey::decode(params, &uvk[..])
+                .map_err(SqliteClientError::CorruptedData)?;
+
+            match account_type {
+                0 => Ok(Some(Account::Zip32(HDSeedAccount(
+                    HDSeedFingerprint::from_bytes(row.get(2)?),
+                    zip32::AccountId::try_from(row.get::<_, u32>(3)?).map_err(|_| {
+                        SqliteClientError::CorruptedData(
+                            "ZIP-32 account ID from db is out of range.".to_string(),
+                        )
+                    })?,
+                    ufvk,
+                )))),
+                1 => Ok(Some(Account::Imported(ImportedAccount::Full(ufvk)))),
+                _ => Err(SqliteClientError::CorruptedData(
+                    "Unrecognized account_type".to_string(),
+                )),
+            }
```

Comment:
Please factor out account type decoding and encoding into a top-level function, as has been done for e.g. `pool_code` etc. or declare constants for the supported `account_type` values, so that we don't have to manually track that the supported set of constants agree for encoding and decoding.

Reply:
I like that idea. I tried to do it in fact before, but rust doesn't let me return `Account::Imported` as a value or type, so I gave up. I'll give it another go though.